src/components/alarm/AlarmController.h

```diff
@@ -38,33 +38,44 @@ namespace Pinetime {
       void SetOffAlarmNow();
       uint32_t SecondsToAlarm();
       void StopAlerting();
-      enum class AlarmState { Not_Set, Set, Alerting };
       enum class RecurType { None, Daily, Weekdays };
       uint8_t Hours() const {
-        return hours;
+        return alarm.hours;
       }
       uint8_t Minutes() const {
-        return minutes;
+        return alarm.minutes;
       }
-      AlarmState State() const {
-        return state;
+      bool IsAlerting() const {
+        return isAlerting;
       }
-      RecurType Recurrence() const {
-        return recurrence;
+      bool IsEnabled() const {
+        return alarm.isEnabled;
       }
-      void SetRecurrence(RecurType recurType) {
-        recurrence = recurType;
+      RecurType Recurrence() const {
+        return alarm.recurrence;
       }
+      void SetRecurrence(RecurType recurrence);
 
     private:
+      static constexpr uint8_t alarmFormatVersion = 1;
```

Comment:
I think limiting us to 128 format versions could bite us in the future. I don't expect to have more than 128 format changes, but I think an additional byte would give us a bit more breathing room

```suggestion
      static constexpr uint16_t alarmFormatVersion = 1;
```

Reply:
I was about to make it 16 bit originally, but don't think it makes sense. Even if the alarm format gets changed in every future release and we assume there will be a new release every month, 8 bit is still enough for over 20 years.

I added a comment to reserve the last value (`255`), so if – for whatever reason – the 254 versions should be exhausted quicker than expected, format version 255 can include a 64 bit extended version field.

Reply:
that sounds reasonable :+1: